js/src/views/Contract/contract.js

```diff
@@ -79,9 +83,17 @@ class Contract extends Component {
     }
   }
 
+  componentWillUnmount () {
+    const { api } = this.context;
+    const { subscriptionId, blockSubscriptionId, contract } = this.state;
+
+    api.unsubscribe('eth_blockNumber', blockSubscriptionId);
+    contract.unsubscribe(subscriptionId);
```

Comment:
Doesn't match with the API change made above. Honestly for conistency sake, should probably just pass the id and let the subscription manager (api) figure out where it sits. We can have a simple additional lookup there. Just makes it much easier from an app developer perspective.